### PR TITLE
Fix segfault for invalid `frozen_string_literals` magic comment

### PR DESCRIPTION
**Pull request by andyw8**
Closes https://github.com/ruby/prism/issues/1629

### CODE REVIEW

**Comment by andyw8** [comment]
(wrong commit, will fix)

File: src/prism.c [hunk 1]
```diff
@@ -5277,10 +5277,10 @@ parser_lex_frozen_string_literal_comment(pm_parser_t *parser) {
...
 
-    size_t key_length = strlen("frozen_string_literal");
+    size_t key_length = strlen("frozen_string_literal:");
     if (key_length > (size_t) (end - cursor)) return;
 
-    const uint8_t *cursor_limit = cursor + (end - cursor) - key_length + 1;
+    const uint8_t *cursor_limit = cursor + (end - cursor) - key_length;
 
     while ((cursor = pm_memchr(cursor, 'f', (size_t) (cursor_limit - cursor), parser->encoding_changed, &parser->encoding)) != NULL) {
         if (memcmp(cursor, "frozen_string_literal", key_length) == 0) {
```
Comment (andyw8) [inline src/prism.c:5286]: This is passing but I'm unsure if the other references to `key_length` are still correct.

Comment (andyw8) [reply src/prism.c:5286]: Ah, it passed locally but I'm seeing CI failures.

**Review by kddnewton — CHANGES_REQUESTED** [review]
Unfortunately you can have space between frozen_string_literal and : so this won't quite work as expected

**Comment by kddnewton** [comment]
Fixed by #1685

**Pull request closed without merging**